### PR TITLE
fix: Update stripe customer email when a user updates cal.com email

### LINKED ISSUE #15232: [CAL-3817] When you change your Cal.com email, it doesn't update Stripe
I updated my Cal.com email and then in Stripe checkout, it's still showing my old email

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3817](https://linear.app/calcom/issue/CAL-3817/when-you-change-your-calcom-email-it-doesnt-update-stripe)</sub>

### PR DESCRIPTION
**Pull request by martinsione**
## What does this PR do?

When a user update their email the change should be reflected on stripe billing portal.

- Fixes https://github.com/calcom/cal.com/issues/15232
- Fixes [CAL-3817]

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] (N/A) - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] (N/A) - I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Update your email and check that the change is reflected in stripe billing portal

### CODE REVIEW

File: apps/web/pages/api/auth/verify-email.ts [hunk 1]
```diff
@@ -101,6 +103,9 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       },
     });
 
+    const stripeCustomerId = await getStripeCustomerIdFromUserId(user.id);
+    await stripe.customers.update(stripeCustomerId, { email: updatedEmail });
```
Comment (joeauyeung) [inline apps/web/pages/api/auth/verify-email.ts:107]: We need to keep self-hosters in mind. We should only make this call if Stripe is enabled. We should also make the stripe customer call only if a stripe customer id is found.

**Review by joeauyeung — DISMISSED** [review]
Thanks for your contribution @martinsione. I have one blocking comment.

File: apps/web/pages/api/auth/verify-email.ts (hunk 1 above)
Comment (martinsione) [reply apps/web/pages/api/auth/verify-email.ts]: Good catch @joeauyeung, could you please re-review when you have a minute?

**Review by Udit-takkar — COMMENTED** [review]
Looks Fine now.

**Review by joeauyeung — APPROVED** [review]
LGTM, thanks again for your contribution.

**Pull request merged**